test/api/models/trial.js

```diff
@@ -300,6 +301,13 @@ describe('Trial', () => {
           .then((trialId) => new Trial({ id: trialId }).fetch({ withRelated: ['documents', 'documents.source'] }))
           .then((trial) => should(trial.toJSON().sources).deepEqual({}))
       ));
+
+      it('returns its own source', () => (
+        factory.create('source', { id: 'nct' })
+          .then(() => factory.create('trialWithRelated', { source_id: 'nct' }))
+          .then((trial) => (should(trial.toJSON().sources).have.key(trial.attributes.source_id))
+          )
+      ));
```

Comment:
Instead of creating a `trialWithRelated` here, use a regular `trial`, otherwise you can't be sure if the source comes from the trial itself or some of its related entities.